core/src/main/java/com/linecorp/armeria/client/endpoint/PropertiesEndpointGroup.java

```diff
@@ -85,13 +93,42 @@ public static PropertiesEndpointGroup of(ClassLoader classLoader, String resourc
     public static PropertiesEndpointGroup of(ClassLoader classLoader, String resourceName,
                                              String endpointKeyPrefix, int defaultPort) {
         validateDefaultPort(defaultPort);
-        return new PropertiesEndpointGroup(loadEndpoints(
+        final URL resourceUrl = getResourceUrl(
                 requireNonNull(classLoader, "classLoader"),
-                requireNonNull(resourceName, "resourceName"),
+                requireNonNull(resourceName, "fileName"));
+        return new PropertiesEndpointGroup(loadEndpoints(
+                resourceUrl,
                 requireNonNull(endpointKeyPrefix, "endpointKeyPrefix"),
                 defaultPort));
     }
 
+    /**
+     * Creates a new {@link EndpointGroup} instance that loads the host names (or IP address) and the port
+     * numbers of the {@link Endpoint} from the {@code resourceName} resource file. The resource file must
+     * contain at least one property whose name starts with {@code endpointKeyPrefix}:. Reloading property files
+     * for changes is {@code false} by default.
+     *
+     * <pre>{@code
+     * example.hosts.0=example1.com:36462
+     * example.hosts.1=example2.com:36462
+     * example.hosts.2=example3.com:36462
+     * }</pre>
+     *
+     * @param resourceName the name of the resource where the list of {@link Endpoint}s is loaded from
+     * @param endpointKeyPrefix the property name prefix
+     * @param defaultPort the default port number to use
+     * @param reloadable whether to watch and reload resource.
+     *
+     * @throws IllegalArgumentException if failed to load any hosts from the specified resource file
+     */
+    public static PropertiesEndpointGroup of(ClassLoader classLoader, String resourceName,
+                                             String endpointKeyPrefix, int defaultPort,
+                                             boolean reloadable) {
```

Comment:
Maybe we can just require `Path` instead of `ClassLoader` and `resourceName` to differentiate this from other factory methods that do not reload? What do you think?